### DISCUSSION TITLE
Questions regarding caching on Android

### DISCUSSION THREAD

**Discussion by IslamRustamov** [TheWidlarzGroup/react-native-video#4281] [Q&A]
### What are you trying to do?

I have a bunch of questions regarding cache processing on Android, that I didn't actually find answers googling

1. In this lib's docs it's saying that cache is going to stay for 30 days and then it will expire. By "expire" we mean it's going to get deleted? Is there a way to decrease this time from 30 days to other value? (Gladly we have cache evictors at least)
2. As much as I understand, cache is getting stored in the device's memory, not in RAM. Is that correct? Does it mean that if I allocate 1gb of cache - my application will not take up M + 1 gb RAM memory? (M is amount of memory app is using)
3. I saw a comment in RVNSimpleCache where a comment was left regarding when to clear the cache https://github.com/TheWidlarzGroup/react-native-video/blob/d45300270eb32e76b24c8d31a5a8c67a07b72b32/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerSimpleCache.kt#L13 . I wonder if releasing cache in destructor is going to work, isn't it? (since we can't really catch the event when our app is getting closed, not sure that destructor is going to do the trick)

### Platforms

Android

**Comment by freeboub** [comment]
The best answer would be that we use exoplayer [SimpleCache](https://developer.android.com/reference/androidx/media3/datasource/cache/SimpleCache) but let's try to answer:
1. I don't know where you found this 30 days timing, but that's not my understanding ... I cache stays forever
2. No cache is on file system, your app will not take 1GB of RAM
3. releasing the cache in destructor is not the expected behavior. The comment is more to add a new api function to allow clearing the cache. Today, it is never cleared (except when it is full, the oldest data will be cleared)

**Reply by IslamRustamov** [reply]
https://docs.thewidlarzgroup.com/react-native-video/other/caching 
![Screenshot 2024-11-08 at 21 08 23](https://github.com/user-attachments/assets/cd43b98d-4229-4820-b850-b8e8c8b685f1)

**Reply by IslamRustamov** [reply]
Thanks for the insight! I will think of better cache clearing approach in my project and if I come up with some cool ideas maybe I will share them here.

**Reply by freeboub** [reply]
> https://docs.thewidlarzgroup.com/react-native-video/other/caching 
> ![Screenshot 2024-11-08 at 21 08 23](https://github.com/user-attachments/assets/cd43b98d-4229-4820-b850-b8e8c8b685f1)
> 

Yep, but this is ios only !

**Reply by IslamRustamov** [reply]
I see, thanks for that
